### PR TITLE
Add more functionality in Visual Studio extension

### PR DESCRIPTION
**Pull request by dthaler**
- Add a host C file when importing an EDL into an app project
- Prompt to select path for the ARM TA Dev Kit when creating an enclave project
- Make ARM platform set _ARM_ preprocessor define for use by the nuget package
- Various fixes to the optee files for the enclave

### CODE REVIEW

File: new_platforms/vsextension/ItemTemplates/oehost/host.vstemplate [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<VSTemplate Version="3.0.0" xmlns="http://schemas.microsoft.com/developer/vstemplate/2005" Type="Item">
+  <TemplateData>
+    <DefaultName>Open Enclave Host Code.c</DefaultName>
```
Comment (HernanGatta) [inline new_platforms/vsextension/ItemTemplates/oehost/host.vstemplate]: Is the trailing `.c` intentional?

File: new_platforms/vsextension/ProjectTemplates/oeenclave/sub.mk [hunk 2]
```diff
@@ -1,30 +1,35 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-include $(NEW_PLATFORMS_PATH)oe_sub.mk
 
-global-incdirs-y += $(NEW_PLATFORMS_PATH)Inc
-global-incdirs-y += optee
+OE_SDK_PATH=../../packages/openenclave.0.2.0-CI-20190123-030613
+OE_SDK_INC_PATH=$(OE_SDK_PATH)/build/native/include
+OEEDGER8R=$(OE_SDK_PATH)/tools/oeedger8r.exe
```
Comment (HernanGatta) [inline new_platforms/vsextension/ProjectTemplates/oeenclave/sub.mk]: The new version of the NuGet package I'm working on will have the Linux version of `oeedger8r` which doesn't have an extension. That way we can ensure this Makefile can also be consumed from Linux and not only WSL.

File: new_platforms/docs/visualstudio_dev.md [hunk 3]
```diff
@@ -42,48 +46,22 @@ flag when creating an enclave to automatically get the OP-TEE like behavior
...
-by creating and using a OPTEE-Simulation-Debug configuration in Visual Studio,
-as follows...
-
-1. Create a new configuration (say, for x86 and called OPTEE-Simulation-Debug)
-based on the Debug configuration. A new configuration can be created
-inside Visual Studio by right clicking on the solution, and accessing
-the "Configuration Manager" screen.
-2. In the "Configuration Properties->"Debugging", make the working directory
-be the directory your enclave is built in (usually "$(OutDir)").
+by creating and using the OPTEE-Simulation-Debug configuration in Visual Studio.
```
Comment (HernanGatta) [inline new_platforms/docs/visualstudio_dev.md]: `ImportEnclaveCommand.cs:Execute` appears to create the `OPTEE-Simulation-Debug` configuration.

File: new_platforms/vsextension/ItemTemplates/oehost/host.vstemplate (hunk 1 above)
Comment (dthaler) [reply new_platforms/vsextension/ItemTemplates/oehost/host.vstemplate]: yes, it was spit out by the template wizard.   If you do Add->Item... to the project it's the string filled into the dialog to save as.  I changed it to enclave_host.c.   But when you add it via by Import Enclave button, the filename is $enclavename$_host.c where $enclavename$ is the base name of the EDL you're importing.

File: new_platforms/vsextension/ProjectTemplates/oeenclave/sub.mk (hunk 2 above)
Comment (dthaler) [reply new_platforms/vsextension/ProjectTemplates/oeenclave/sub.mk]: removed .exe

File: new_platforms/vsextension/ProjectTemplates/oeenclave/sub.mk [hunk 4]
```diff
@@ -1,30 +1,35 @@
...
+CFLAGS +=                              \
+    -I..                               \
+    -I$(OE_SDK_INC_PATH)/new_platforms \
+    -I$(OE_SDK_INC_PATH)
 
-../$projectname$_t.h: ../../$projectname$.edl
-	$(OEEDGER8R) --trusted --trusted-dir .. --search-path "$(NEW_PLATFORMS_PATH)Inc$(OEPATHSEP)$(OE_SDK_ROOT_PATH)include" ../../$projectname$.edl
+CFLAGS += -DLINUX -DOE_USE_OPTEE
+
+libdirs += $(OE_SDK_PATH)/lib/native/gcc6/optee/v3.3.0/vsexpress-qemu_armv8
```
[the remark this answers is not in the document]
Comment (dthaler) [reply new_platforms/vsextension/ProjectTemplates/oeenclave/sub.mk]: removed 's', added 'a' as requested

File: new_platforms/docs/visualstudio_dev.md [hunk 5]
```diff
@@ -12,7 +12,11 @@ on both SGX and OP-TEE, and use it from your own application.
 2. To develop for OP-TEE, install the [Prerequisites for OP-TEE](ta_debugging_wsl.md)
 
 **To create your own enclave:**
-1. In Visual Studio, add a new Visual C++ "Open Enclave TEE Project".
+1. In Visual Studio, add a new Visual C++ "Open Enclave TEE Project".  If you want
+to build for OP-TEE (whether in addition to, or instead of, SGX) you will need
+to select the path to the ta_dev_kit.mk file in the output of an OP-TEE build.
+For example, this might be in a optee_os\out\arm-plat-vexpress\export-ta_arm32\mk
```
[the remark this answers is not in the document]
Comment (dthaler) [reply new_platforms/docs/visualstudio_dev.md]: Do we just have 64-bit in the nuget or also 32-bit?

File: new_platforms/docs/visualstudio_dev.md (hunk 3 above)
Comment (dthaler) [reply new_platforms/docs/visualstudio_dev.md]: Updated

File: new_platforms/docs/visualstudio_dev.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (HernanGatta) [reply new_platforms/docs/visualstudio_dev.md]: We have three platforms in the NuGet package; see E-mail.

File: new_platforms/docs/visualstudio_dev.md [hunk 6]
```diff
@@ -42,48 +46,27 @@ flag when creating an enclave to automatically get the OP-TEE like behavior
...
-Visual Studio, also add them to the sub.mk file in your optee subdirectory
-4. In your application project properties, under "Linker"->"Input", add
-rpcrt4.lib to the Additional Dependencies (All Configurations, ARM platform)
-which is required for string-to-UUID conversion, and remove any sgx libs.
-5. In your application project properties, update the Additional Include
-Directories to insert the $(NewPlatformsDir)include\optee\host and
-$(NewPlatformsDir)include\optee paths before the $(NewPlatformsDir)include path that you
-added earlier.
-6. On the destination machine, apply the uuids.reg file
+1. Using [Bash on Ubuntu on Windows](https://docs.microsoft.com/en-us/windows/wsl/about) shell,
```
Comment (HernanGatta) [inline new_platforms/docs/visualstudio_dev.md]: Perhaps: "Using the [Bash on Ubuntu on Windows](...) shell [...]".

File: new_platforms/vsextension/ProjectTemplates/oeenclave/MyEnclave.vcxproj [hunk 7]
```diff
@@ -214,46 +213,35 @@
...
-      <BasicRuntimeChecks>Default</BasicRuntimeChecks>
-      <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
       <Optimization>Disabled</Optimization>
       <WarningLevel>Level3</WarningLevel>
-      <AdditionalIncludeDirectories>$(RIoTDir)CyReP\cyrep;$(OpteeDir)lib\libutee\include;$(OpteeDir)lib\libutils\ext\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <AdditionalLibraryDirectories>$(OutDir);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <EntryPointSymbol>
       </EntryPointSymbol>
```
Comment (HernanGatta) [inline new_platforms/vsextension/ProjectTemplates/oeenclave/MyEnclave.vcxproj]: Empty tag.

File: new_platforms/docs/visualstudio_dev.md (hunk 6 above)
Comment (dthaler) [reply new_platforms/docs/visualstudio_dev.md]: done

File: new_platforms/vsextension/ProjectTemplates/oeenclave/MyEnclave.vcxproj (hunk 7 above)
Comment (dthaler) [reply new_platforms/vsextension/ProjectTemplates/oeenclave/MyEnclave.vcxproj]: that's how the template wizard generated it.   Anyway, removed throughout.

**Pull request merged**